include/operations/blas_operators.hpp

```diff
@@ -71,17 +55,17 @@ struct IndVal {
 @param inital Initial value used in the init function of the oeprator.
 @param expr Return expression of the eval function of the operator.
 */
-#define SYCLBLAS_DEFINE_BINARY_OPERATOR(name, initial, expr)   \
-  struct name {                                                \
-    template <typename L, typename R>                          \
-    static R eval(const L& l, const R& r) {                    \
-      return expr;                                             \
-    }                                                          \
-                                                               \
-    template <typename R>                                      \
-    static typename R::value_type init(const R& r) {           \
-      return constant<typename R::value_type, initial>::value; \
-    }                                                          \
+#define SYCLBLAS_DEFINE_BINARY_OPERATOR(name, initial, expr)          \
+  struct name {                                                       \
+    template <typename L, typename R>                                 \
+    static typename strip_asp<R>::type eval(const L &l, const R &r) { \
```

Comment:
I think @mehdi-goli has almost fixed that and this workaround will disappear shortly, so it makes sense not to touch it, i think.